### ISSUE TITLE
missing parent lookup records were found

### ISSUE THREAD

**Issue by ilanhazan** [forcedotcom/SFDX-Data-Move-Utility#25]
Hello.
I have two objects: Account and Deal__c.
Account is the parent SObject and Deal__c is the child. 
Let's assume that the Account lookup field on the Deal__c is called Company__c.

The export.json configuration:

{
    "objects": [

        {
            "query": "SELECT Id, Name, Company__c FROM Deal__c WHERE Name = 'Ilan Deal1'",
            "operation": "Insert",
            "externalId": "Name",
            "allRecords": false  
        },
    
        {
            "query": "SELECT Id, Name FROM Account WHERE Name = 'Ilan'",
            "operation": "Insert",
            "externalId": "Name"
        }   

    ],
    "promptOnMissingParentObjects": false
}

When I run the command, it failed to do the copy and I get the following:
WARNING: [2020-05-21  10:22:46.826] [WARN] {Deal__c} 1 missing parent lookup records were found. See MissingParentRecordsReport.csv file for the details.
....
WARNING: [2020-05-21  10:22:47.787] [WARN] [Batch# REST:Insert] {Deal__c} Incompleted. 0 records succeeded, 1 records failed.

Note: this error happens also with "operation": "Upsert".
The content of MissingParentRecordsReport.csv is:
Date update,Id,Child ExternalId,Child lookup,Child SObject,Missing value,Parent ExternalId,Parent SObject
2020-05-21  10:22:46.825,a0526000006P9ppAAC,Company__r.Name,Company__c,Deal__c,Ilan,Name,Account

**Comment by hknokh** [comment]
Hello, @ilanhazan 
I checked your data model.
Created the similar data mode to check.
This is my export.json:

{
    "objects": [
        {
            "operation": "Upsert",
            "externalId": "Name",
            "query": "SELECT Id, Name FROM Account WHERE Name = 'ACC_10000'"
        },
        {
            "operation": "Upsert",
            "externalId": "Name",
            "query": "SELECT Id, Account__c, Name FROM TestObject__c WHERE Name = 'test object name'",
			"allRecords": false "
        }
	],
	"promptOnMissingParentObjects": false
}
- The target org is empty or not empty. 
- Tried both Upsert and Insert operations.
-- Tried several variations of WHERE clause to select existing or missing records.
In all tested variations it is working perfectly without any extra warnings. Also re-checked the program logic - should not be any warning in this case. 
In summary I can't reproduce and debug your issue hence unfortunately so far have no solution for your issue.
I need to see what happens on your local machine but it's impossible.
Of course such issues also can depend on variety of things for example which environment component are installed like nodejs, npm, sfdc-cli etc.
If you can debug it yourself and tell me what is the problem it will be much appreciated :) 
For this purpose you can run plugin using it's source on Github instead of installing it with sfdx.
Thank you!

**Comment by Aravindh04** [comment]
@hknokh Can you please provide the resolution comments?

**Comment by hknokh** [comment]
Hello @Aravindh04 
Unfortunatelly I have no resolution since can't reproduce on my end as I already commented above.
Please debug it yourself on your side. 
Or you can give me an access to your Source and Target test envs where I can repdocued the issue in live mode. Perhaps I can check it for you there.

**Comment by hknokh** [comment]
Hi, 
I'm closing the case for now.
Keep me informed if you will have an update.
Thanks
